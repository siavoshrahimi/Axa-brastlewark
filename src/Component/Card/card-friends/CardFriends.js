import React from 'react';
import {Link} from "react-router-dom";

import './CardFreinds.scss';

const CardFriends = ({friend}) =>{
    return(
        <Link to={{pathname:`/detail/${friend.name}`,state:{friend}}}
              title={friend.name}
              key={friend.id}>
            <img className='friends-img' src={friend.thumbnail} alt={friend.name}/>
        </Link>
    )
}

export default CardFriends;