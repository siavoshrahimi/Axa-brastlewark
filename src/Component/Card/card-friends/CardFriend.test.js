import React from 'react';
import {shallow} from 'enzyme';

import CardFriends from './CardFriends';


it('expect to render card detail component' ,() =>{
    const mockFriends = {
        id:1,
        name:'david',
        thumbnail: "marc",
    }
    const wrapper = shallow(<CardFriends friend={mockFriends} />);
    expect(wrapper).toMatchSnapshot();
})