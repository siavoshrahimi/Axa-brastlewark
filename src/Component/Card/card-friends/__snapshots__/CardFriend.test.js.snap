// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card detail component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardFriends
    friend={
      Object {
        "id": 1,
        "name": "david",
        "thumbnail": "marc",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "class",
    "props": Object {
      "children": <img
        alt="david"
        className="friends-img"
        src="marc"
      />,
      "title": "david",
      "to": Object {
        "pathname": "/detail/david",
        "state": Object {
          "friend": Object {
            "id": 1,
            "name": "david",
            "thumbnail": "marc",
          },
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "david",
        "className": "friends-img",
        "src": "marc",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "class",
      "props": Object {
        "children": <img
          alt="david"
          className="friends-img"
          src="marc"
        />,
        "title": "david",
        "to": Object {
          "pathname": "/detail/david",
          "state": Object {
            "friend": Object {
              "id": 1,
              "name": "david",
              "thumbnail": "marc",
            },
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "david",
          "className": "friends-img",
          "src": "marc",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
